stock_request_picking_type/models/stock_request_order.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Open Source Integrators
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequestOrder(models.Model):
+    _inherit = "stock.request.order"
+
+    @api.model
+    def _get_default_picking_type(self):
+        companies = self._context.get("allowed_company_ids", []).copy()
+        companies.append(False)
+        return (
+            self.env["stock.picking.type"]
+            .search(
+                [
+                    ("code", "=", "stock_request_order"),
+                    "|",
+                    ("warehouse_id.company_id", "in", companies),
+                    ("warehouse_id", "=", False),
+                ],
+                limit=1,
+            )
+            .id
+        )
+
+    picking_type_id = fields.Many2one(
+        comodel_name="stock.picking.type",
+        string="Operation Type",
+        default=_get_default_picking_type,
+        required=True,
+    )
+
+    @api.onchange("warehouse_id")
+    def onchange_warehouse_picking_id(self):
```

Comment:
This should now be avoided. You can change the picking_type_id field to a computed/stored/readonly=False field instead and change this to _compute method.